Objects/codeobject.c

```diff
@@ -885,14 +887,17 @@ _PyCode_SetExtra(PyObject *code, Py_ssize_t index, void *extra)
             co_extra->ce_extras, tstate->co_extra_user_count * sizeof(void*));
```

Comment:
Previously allocated `co_extra->ce_extras` is leaked if `PyMem_Realloc()` returns `NULL`. It would be better to save the result of `PyMem_Realloc()` to the variable and check it for `NULL` first than assign it to `co_extra->ce_extras`.